lib/core/Canvas.js

```diff
@@ -307,8 +308,8 @@ Canvas.prototype.getLayer = function(name, index) {
  */
 Canvas.prototype._createLayer = function(name, index) {
 
-  if (!index) {
-    index = 0;
+  if (typeof index === 'undefined') {
```

Comment:
What exactly was broken and can we have a test case for that?

Reply:
`0` is falsey, so we changed the index in cases where index was defined. This was okay as long as we replaced it with a 0 again, but with this PR we replace it with a 1

Reply:
I'm not sure if we need an explicit test case for this, as the existing ones already blew up when the index was implicitly changed

Reply:
> What exactly was broken and can we have a test case for that?

I referred to the general case. This PR changes something, right? What does it change (improve for the better)? I'd love that to be asserted in a test case.

Reply:
Understood :+1:
I made some existing tests more explicit and added some to ensure base layer and planes behave the same